src/Http/GraphRequest.php

```diff
@@ -231,17 +231,17 @@ public function setTimeout($timeout)
     public function execute($client = null)
     {
         if (is_null($client)) {
-            $client = $this->createGuzzleClient($this->proxyPort);
+            $client = new Client();
         }
 
         $result = $client->request(
             $this->requestType, 
             $this->_getRequestUrl(), 
-            [
+            array_merge($this->getGuzzleOptions($this->proxyPort), [
```

Comment:
Don't really understand this comment.
We need to assume that people are using a compliant client that works with the SDK.

Since the http client interface is not standardize (like PSR7 for example), we need to declare the current implementation of a http client we are using.

From my point of view, we need to force an http client (Guzzle or HTTPlug or another one), the simpliest is Guzzle : 

```php 
use GuzzleHttp\ClientInterface;

    /**
    * Executes the HTTP request using Guzzle
    *
    * @param ClientInterface $client The client to use in the request
    *
     * @throws GraphException if response is invalid
     *
    * @return mixed object or array of objects
    *         of class $returnType
    */
    public function execute(ClientInterface $client = null)
    {
```

If the people use HTTPlug, they can use the guzzle adapter to give to msgraph-sdk the client needed.

Or if you want to use [HTTPlug to abstract completely](http://docs.php-http.org/en/latest/httplug/library-developers.html), it's not the same work since it need to refractor the whole sdk since there is a lot of dependency with Guzzle (also on generated part on the sdk) and it will be breaking change and a major version bump

Reply:
I think we can work with Guzzle as a default for now as we prep for support PSR-18 HTTP clients in the future.